packages/notifier/src/notifier.js

```diff
@@ -51,35 +51,26 @@ export const makeNotifier = sharableInternalsP => {
  * @returns {NotifierRecord<T>} the notifier and updater
  */
 export const makeNotifierKit = (...initialStateArr) => {
-  /** @type {PromiseRecord<UpdateRecord<T>>|undefined} */
-  let optNextPromiseKit;
-  /** @type {UpdateCount} */
-  let currentUpdateCount = 1; // avoid falsy numbers
-  /** @type {UpdateRecord<T>|undefined} */
-  let currentResponse;
-
-  const hasState = () => currentResponse !== undefined;
-
-  const final = () => currentUpdateCount === undefined;
+  const { publisher, subscriber } = makeEmptyPublishKit();
 
+  /**
+   * @template T
+   * @type {BaseNotifier<T>}
+   */
   const baseNotifier = Far('baseNotifier', {
-    // NaN matches nothing
-    getUpdateSince(updateCount = NaN) {
-      if (
-        hasState() &&
-        (final() ||
-          (currentResponse && currentResponse.updateCount !== updateCount))
-      ) {
-        // If hasState() and either it is final() or it is
-        // not the state of updateCount, return the current state.
-        assert(currentResponse !== undefined);
-        return Promise.resolve(currentResponse);
-      }
-      // otherwise return a promise for the next state.
-      if (!optNextPromiseKit) {
-        optNextPromiseKit = makePromiseKit();
+    getUpdateSince(sinceUpdateCount = undefined) {
+      // TODO: Narrow to exclude number.
+      if (typeof sinceUpdateCount === 'number') {
+        sinceUpdateCount = BigInt(sinceUpdateCount);
       }
-      return optNextPromiseKit.promise;
+      return E.when(
+        subscriber.subscribeAfter(sinceUpdateCount),
+        ({ head: { value, done }, publishCount }) => {
+          // Unlike publish kit, notifier reports no count with a final value.
+          const updateCount = done ? undefined : publishCount;
+          return harden({ value, updateCount });
+        },
+      );
```

Comment:
This implementation is the source of the incompatibilities.  It needs to have the same semantics, which is (sketching it out):

```js
  let currentUpdateCount = 1; // avoid falsy numbers
  let nextValueP;
  let currentValueP;
  const iterator = subscribeLatest(subscriber)[Symbol.asyncIterator]();
  const getNextValueP  = () => {
      if (!nextValueP) {
        nextValueP = iterator.next().then(({done, value}) => {
           nextValueP = undefined;
           currentValueP = Promise.resolve(value);
           currentUpdateCount += 1;
           return currentValue;
        });

  const baseNotifier = Far('baseNotifier', {
    // NaN matches nothing
    getUpdateSince: async (updateCount = NaN) {
      if (!updateCount || updateCount <= currentUpdateCount) {
        if (!currentValueP) {
          currentValueP = getNextValueP();
        }
        return currentValueP;
      }
      return getNextValueP();
    },
  });
```